### DISCUSSION TITLE
My zigbee2mqtt installation wont update using /opt/zigbee2mqtt/update.sh

### DISCUSSION THREAD

**Discussion by FoxConsult** [Koenkk/zigbee2mqtt#13672] [General]
Everything works fine but when running the update script /opt/zigbee2mqtt/update.sh, I get an "Already up to date." despite new package.json and package-lock.json files in repo https://github.com/Koenkk/zigbee2mqtt/tree/dev so I decided to reinstall zigbee2mqtt:

Made a copy of the Data directory.
Deleted /opt/zigbee2mqtt
Ran the installation as per https://www.zigbee2mqtt.io/guide/installation/, "Installing"
Deleted /opt/zigbee2mqtt/Data
Restored /opt/zigbee2mqtt/Data from step one above
Executed sudo systemctl start zigbee2mqtt

A git checkout dev shows I'm on the development branch:

**************************
pi@raspberrypi:/opt/zigbee2mqtt $ git checkout dev
M       data/configuration.yaml
Already on 'dev'
**************************

Output from running sh update.sh:

**************************
pi@raspberrypi:/opt/zigbee2mqtt $ sh update.sh
Stopping Zigbee2MQTT...
Creating backup of configuration...
Updating...
Already up to date.
Installing dependencies...

added 827 packages, and audited 828 packages in 2m

81 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Restore configuration...
Starting Zigbee2MQTT...
Done!
**************************

Any suggestions how I can correct this?

**Comment by Koenkk** [comment]
What is the output of `git log`?

**Reply by FoxConsult** [reply]
pi@raspberrypi:/opt/zigbee2mqtt $ git log
commit e0a9323d6efd0bee5547e537588010aece93e0c9 (grafted, HEAD -> dev, origin/master, origin/HEAD, master)
Author: Koen Kanters <koenkanters94@gmail.com>
Date:   Fri Aug 19 20:57:44 2022 +0200

    Update dont-stale label
pi@raspberrypi:/opt/zigbee2mqtt $

**Reply by Koenkk** [reply]
Please make a backup of your `data` folder. Next execute:

```bash
git fetch origin
git reset --hard origin/dev
npm ci
```

Restore your `data` folder and start z2m as usual

**Reply by FoxConsult** [reply]
Thank you for the quick reply.

I followed your proposed steps but got an error when running "git reset --hard origin/dev"

pi@raspberrypi:~ $ cd /opt/zigbee2mqtt
pi@raspberrypi:/opt/zigbee2mqtt $ sudo systemctl stop zigbee2mqtt

Made a copy of /opt/zigbee2mqtt/data 

pi@raspberrypi:/opt/zigbee2mqtt $ git fetch origin
pi@raspberrypi:/opt/zigbee2mqtt $ git reset --hard origin/dev
fatal: ambiguous argument 'origin/dev': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

pi@raspberrypi:/opt/zigbee2mqtt $ git branch -r
  origin/HEAD -> origin/master
  origin/master

Looks like my system can't see the remote branch "dev"?

**Reply by Koenkk** [reply]
What is the output of `cat /opt/zigbee2mqtt/.git/config`?

**Reply by FoxConsult** [reply]
I decided to make a copy of /opt/zigbee2mqtt/data and deleted /opt/zigbee2mqtt. I then installed the dev branch of your project using git clone --branch dev --depth 1 https://github.com/Koenkk/zigbee2mqtt.git /opt/zigbee2mqtt.

The output of cat /opt/zigbee2mqtt/.git/config:

[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "origin"]
        url = https://github.com/Koenkk/zigbee2mqtt.git
        fetch = +refs/heads/dev:refs/remotes/origin/dev
[branch "dev"]
        remote = origin
        merge = refs/heads/dev

As soon as you check in any new files to the repository, I will try the update.sh script again.

**Reply by FoxConsult** [reply]
After installing using git clone --branch dev --depth 1 https://github.com/Koenkk/zigbee2mqtt.git /opt/zigbee2mqtt the update.sh script now updates my dev branch installation.

/Tommie

**Reply by FoxConsult** [reply]
Unfortunately, I am getting the same error updating the dev branch as I had shown in this thread (#13672).
I have not made any changes to my installation; no new devices, no restoring backups, just running sh update.sh as soon as https://github.com/Koenkk/zigbee2mqtt/tree/dev had some updates.

Two days ago I started getting "Already up to date." when running sh update.sh:

/opt/zigbee2mqtt $ sh update.sh
Stopping Zigbee2MQTT...
Creating backup of configuration...
Updating...
Already up to date.
Installing dependencies...


So, as per instructions in this thread, I did "Please make a backup of your data folder. Next execute:

git fetch origin
/opt/zigbee2mqtt $ git reset --hard origin/dev

fatal: ambiguous argument 'origin/dev': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

Here is the output of cat /opt/zigbee2mqtt/.git/config:

[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "origin"]
        url = https://github.com/Koenkk/zigbee2mqtt.git
        fetch = +refs/heads/master:refs/remotes/origin/master
[branch "master"]
        remote = origin
        merge = refs/heads/master
